include/swift/Basic/SymbolicLinks.h

```diff
@@ -0,0 +1,30 @@
+//===--- SymbolicLinks.h ----------------------------------------*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef SWIFT_BASIC_SYMBOLICLINKS_H
+#define SWIFT_BASIC_SYMBOLICLINKS_H
+
+#include "swift/Basic/LLVM.h"
+#include "llvm/Support/VirtualFileSystem.h"
+#include <string>
+
+namespace swift {
+
+/// Tries to resolve symbolic links in a given path, but preserves
+/// substitute drives on Windows to avoid MAX_PATH issues.
+/// On failure, returns the original path.
+std::string resolveSymbolicLinks(llvm::StringRef InputPath,
+    llvm::vfs::FileSystem *FileSystem);
```

Comment:
I don't expect you to fix this, but it kind of seems like this is something that `FileSystem` should have.

Reply:
I'm torn on this. It has `getRealPath`, which is the real filesystem operation. This logic that has a special case for Windows is a higher-level construct, so it may make sense to keep it separate.